R/fcast.R

```diff
@@ -165,11 +165,7 @@ dcast.data.table <- function(data, formula, fun.aggregate = NULL, sep = "_", ...
       # tryCatch(fill.default <- dat[0L][, eval(fun.call)], error = function(x) stop(errmsg, call.=FALSE))
       if (nrow(fill.default) != 1L) stop(errmsg, call.=FALSE)
     }
-    if (!any(valnames %chin% varnames)) {
-      dat = dat[, eval(fun.call), by=c(varnames)]
-    } else {
-      dat = dat[, { .SD; eval(fun.call) }, by=c(varnames), .SDcols = valnames]
```

Comment:
when this branch was written, the branch at 131:

```
if (length(dupidx)) {
```

didn't exist. Now that it's there, the names are already unique by the time they reach this branch, so only `!any()` is possible.